### PR TITLE
Add install_strategy option to bootstrap install scripts

### PR DESCRIPTION
**Pull request by wrightp**
This PR is part of JEX-643.

When `install_strategy` is set to `once` the install scripts will not install packages when the product is already installed (validated by root paths). All other conditions behave in the current manner which is to always install.

### CODE REVIEW

File: acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+execute "bundle exec ruby -e \"require 'mixlib/install'; puts Mixlib::Install.new(product_name: 'chef', product_version: :latest, channel: :stable, install_command_options: {install_strategy: 'once'}).install_command\" > ../.acceptance_data/ubuntu_install_command_once.sh" do
```
Comment (scotthain) [inline acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb]: Can we maybe do a heredoc or something just to make this easier to read?

File: acceptance/windows-server-2012r2-once/.acceptance/acceptance-cookbook/recipes/provision.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+execute "bundle exec ruby -e \"require 'mixlib/install'; puts Mixlib::Install.install_ps1, 'install -install_strategy once'\" > ../.acceptance_data/powershell_install_once.ps1" do
```
Comment (scotthain) [inline acceptance/windows-server-2012r2-once/.acceptance/acceptance-cookbook/recipes/provision.rb]: Same with heredoc?

File: acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb (hunk 1 above)
Comment (wrightp) [inline acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb]: You're not supposed to look at that @scotthain❗️

Comment (scotthain) [inline acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb]: 🤣

File: lib/mixlib/install/generator/powershell/scripts/install_project.ps1 [hunk 3]
```diff
@@ -48,15 +48,25 @@ function Install-Project {
...
     # SHA256 checksum of the download file
     # Must be present when using download_url_override
+    [string]
+    $checksum,
+    # Set to 'once' to skip install if project is detected
+    [string]
+    $install_strategy
   )
 
+  if ((Test-Path "$env:systemdrive\opscode\$project\embedded") -and ($install_strategy -eq 'once')) {
```
Comment (coderanger) [inline lib/mixlib/install/generator/powershell/scripts/install_project.ps1]: Is there a reason for the mismatch on which path is checked on bash vs. powershell (root vs root/embedded)?

File: acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb (hunk 1 above)
Comment (wrightp) [inline acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb]: I added a generic comment to all provision recipes

File: lib/mixlib/install/generator/powershell/scripts/install_project.ps1 (hunk 3 above)
Comment (wrightp) [inline lib/mixlib/install/generator/powershell/scripts/install_project.ps1]: @thommay Do you recall if there was a specific reason for the difference in path checking for
[ps1](https://github.com/chef/mixlib-install/blob/master/support/install_command.ps1#L2) and [shell](https://github.com/chef/mixlib-install/blob/master/support/install_command.sh#L207)?

File: acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb (hunk 1 above)
Comment (scotthain) [inline acceptance/ubuntu_install_command_once/.acceptance/acceptance-cookbook/recipes/provision.rb]: Ok I'll allow it ;)

**Pull request merged**